repose-aggregator/components/services/container-configuration-service/impl/src/main/scala/org/openrepose/nodeservice/containerconfiguration/ContainerConfigurationServiceImpl.scala

```diff
@@ -68,8 +69,9 @@ class ContainerConfigurationServiceImpl @Inject()(@Value(ReposeSpringProperties.
     Optional ofNullable {
       Option(patchedDeploymentConfiguration.getViaHeader)
         .map(_.getRequestPrefix)
+        .filter(StringUtils.isNotBlank)
         // TODO for v9.0.0.0: Remove the .orElse().
-        .orElse(Option(patchedDeploymentConfiguration.getVia))
+        .orElse(Option(patchedDeploymentConfiguration.getVia).filter(StringUtils.isNotBlank))
```

Comment:
Couldn't you just `filter` after the `orElse` so that you only have to do it once?